main.go

```diff
@@ -6,224 +6,43 @@
-		}
-
-		if getFlags.NArg() > 1 {
-			exitOnUsageError(flags.Usage, "Too many arguments except none or binary/package ")
-		}
-
-		target := getFlags.Arg(0)
-		if *getRename != "" && *getName != "" {
-			exitOnUsageError(flags.Usage, "Both -n and -r were specified. You can either rename or create new one.")
-		}
-		if *getName != "" && !regexp.MustCompile(`[a-zA-Z0-9.-_]+`).MatchString(*getName) {
-			exitOnUsageError(flags.Usage, *getName, "-n name contains not allowed characters")
-		}
-		if *getRename != "" && !regexp.MustCompile(`[a-zA-Z0-9.-_]+`).MatchString(*getRename) {
-			exitOnUsageError(flags.Usage, *getRename, "-r name contains not allowed characters")
-		}
-
-		cmdFunc = func(ctx context.Context, r *runner.Runner) (err error) {
-			relModDir := *getModDir
-			modDir, err := filepath.Abs(relModDir)
-			if err != nil {
-				return errors.Wrap(err, "abs")
-			}
-			defer func() {
-				if err == nil {
-					// Leave tmp files on error for debug purposes.
-					if cerr := cleanGoGetTmpFiles(modDir); cerr != nil {
-						logger.Println("cannot clean tmp files", err)
-					}
-				}
-			}()
-
-			cfg := getConfig{
-				runner:    r,
-				modDir:    modDir,
-				relModDir: relModDir,
-				name:      *getName,
-				rename:    *getRename,
-				verbose:   *verbose,
-				link:      *getLink,
-			}
-
-			if err := get(ctx, logger, cfg, target); err != nil {
-				return errors.Wrap(err, "get")
-			}
-
-			pkgs, err := bingo.ListPinnedMainPackages(logger, modDir, true)
-			if err != nil {
-				return errors.Wrap(err, "list pinned")
-			}
-			if len(pkgs) == 0 {
-				return bingo.RemoveHelpers(modDir)
-			}
-			return bingo.GenHelpers(relModDir, version.Version, pkgs)
-		}
-	case "list":
-		listFlags.SetOutput(os.Stdout)
-		if err := listFlags.Parse(flags.Args()[1:]); err != nil {
-			exitOnUsageError(flags.Usage, "Failed to parse flags for get command:", err)
-		}
-
-		if !*verbose && *listVerbose {
-			*verbose = true
-		}
-
-		if *listModDir == "" {
-			exitOnUsageError(flags.Usage, "'moddir' flag cannot be empty")
-		}
-
-		if listFlags.NArg() > 1 {
-			exitOnUsageError(flags.Usage, "Too many arguments; only one binary/package or no argument is expected ")
-		}
-
-		target := listFlags.Arg(0)
-		cmdFunc = func(ctx context.Context, r *runner.Runner) error {
-			modDir, err := filepath.Abs(*listModDir)
-			if err != nil {
-				return errors.Wrap(err, "abs")
-			}
-			pkgs, err := bingo.ListPinnedMainPackages(logger, modDir, false)
-			if err != nil {
-				return err
-			}
-
-			bingo.SortRenderables(pkgs)
-			return pkgs.PrintTab(target, os.Stdout)
-		}
-	case "version":
-		cmdFunc = func(ctx context.Context, r *runner.Runner) error {
-			_, err := fmt.Fprintln(os.Stdout, version.Version)
-			return err
-		}
-	default:
-		exitOnUsageError(flags.Usage, "No such command", flags.Arg(0))
-	}
-
-	g := &run.Group{}
-	g.Add(run.SignalHandler(context.Background(), syscall.SIGINT, syscall.SIGTERM))
```

Comment:
This allowed us to Ctrl+C and gracefully finish our CLI, I wonder if it works well now.

Reply:
It seems that I have omitted the logic of this part